native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ListenerUtils.java

```diff
@@ -194,4 +196,17 @@ public static void stopObserverContext(Environment environment, BObject context)
             environment.setStrandLocal(KEY_OBSERVER_CONTEXT, parentContext);
         }
     }
+  
+    public static void stopObserverContextWithError(Environment environment, BObject context, BError error) {
+        ObserverContext observerContext = (ObserverContext) context.getNativeData(KEY_OBSERVER_CONTEXT);
+        if (observerContext != null && observerContext.isManuallyClosed()) {
+            if (error != null) {
+                observerContext.addProperty(PROPERTY_ERROR_VALUE, error);
+            }
```

Comment:
Can this be null? Ballerina API only accepts `error`, so this can't be a null value. Therefore, this check is redundant.

Reply:
I will remove this check.